multiplex.go

```diff
@@ -167,6 +167,13 @@ func (mp *Multiplex) NewStream() (*Stream, error) {
 
 func (mp *Multiplex) NewNamedStream(name string) (*Stream, error) {
 	mp.chLock.Lock()
+
+	// We could call IsClosed but this is faster (given that we already have
+	// the lock).
+	if mp.channels == nil {
```

Comment:
Meh. I've just commented it and left it as-is.

There's no reason to but the fact that `channels == nil` means the connection is closed is non-obvious. I'm probably just paranoid.